src/main/scala/mastermind/Mastermind.scala

```diff
@@ -13,19 +13,36 @@ val validColors = List('B', 'Y', 'R', 'G')
 
 /** Get a random color from the list of valid colors */
 def getRandomColor(): Color =
-  ???
+  Random.shuffle(validColors).head
 
 /** Given four colors, make a board from them */
 def makeBoardFromColors(c1: Color, c2: Color, c3: Color, c4: Color): Board =
-  ???
+  List(c1, c2, c3, c4).mkString("")
 
 /** Create a random board */
 def getRandomBoard(): Board =
-  ???
+  makeBoardFromColors(getRandomColor(), getRandomColor(), getRandomColor(), getRandomColor())
 
 /** Play one round of the game */
 def playRound(board: Board): (Int, Int) =
-  ???
+  var firstGuess: Char = 'a'
```

Comment:
I did this in a very different way than you did. I simply wrote:
println("Enter a guess for spot 1: ")
  val guess1 = scala.io.StdIn.readLine()
It is much more simple than your solution but it lacks the ability to check if an input is a valid color which I really liked about your solution.